### DISCUSSION TITLE
Better support for Prisma Schema Extensions

### DISCUSSION THREAD

**Discussion by molomby** [keystonejs/keystone#8677] [Feature requests]
Prisma have released (in v4.7.0, 20 June 2023) a new mechanism for [adding extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions) to the Prisma client, queries, models and results.

Implementing an extension involves wrapping the Prisma client. In theory, this is possible in Keystone already, by replacing the default scripts (`start`, `dev`, etc.) but we can probably do better. We should be able to add a fairly trivial hook to the root Keystone config to make this much easier for devs.

**Comment by demsey2** [comment]
on top of that, there is a new extension that allows you to setup read replicas https://www.prisma.io/blog/read-replicas-prisma-client-extension-f66prwk56wow

That extension requires Prisma 5.2
